node/libs/protobuf/src/proto_fmt.rs

```diff
@@ -282,14 +288,22 @@ pub fn encode_json<T: ProtoFmt>(x: &T) -> String {
     encode_json_proto(&x.build())
 }
 
+/// Implementation of serde::Deserialize for arbitrary ReflectMessage.
+pub fn serde_deserialize<'de, T: ReflectMessage + Default, D: serde::Deserializer<'de>>(
+    d: D,
+) -> Result<T, D::Error> {
+    let mut p = T::default();
+    let msg = prost_reflect::DynamicMessage::deserialize(p.descriptor(), d)?;
+    p.merge(msg.encode_to_vec().as_slice()).unwrap();
```

Comment:
Nit: Is there a particular reason not to use [`DynamicMessage::transcode_to()`](https://docs.rs/prost-reflect/latest/prost_reflect/struct.DynamicMessage.html#method.transcode_to) here? It [seems to implement](https://docs.rs/prost-reflect/latest/src/prost_reflect/dynamic/mod.rs.html#576-582) the same logic under the hood.

Reply:
It's a microoptimization, in a sense that I'm reusing the already existing instance p, which I had to create to call descriptor(). Actually I was thinking about getting rid of ReflectMessage trait altogether in favor of defining our own, so that descriptor can be a class method (not taking &self as an argument).